### PR TITLE
Fix issues rebasing hotfixes

### PR DESCRIPTION
**Pull request by brandoncollins7**

### CODE REVIEW

**Comment by saenyakorn** [comment]
Thanks for contribution! I would take some of my free time on testing this. I'll let you know if I've done :). As of now, I concerned that would "force rebase" work on every use cases. That's what I'll test.

**Comment by saenyakorn** [comment]
Sorry for late! I'm a bit busy lately. I have test it. Here's my mock experiments

Common scenario: Let say that I have one package to develop named `docs`. It's start with version `0.0.0`

**Scenario 1**: `docs` was pre-release with version `0.0.1-beta0`. But the team found that version `0.0.0` need a hotfix. 

1. Branch from `main` branch named `hotfix`
2. Fix the issue and commit it ti branch `hotfix`
3. Add changeset file as always to describe what you change for this branch
4. Open the PR, wait for approvals, and then merge it
5. The PR `Version Packages` appeared. Merge it
6. `docs` was bumped to version `0.0.1`. The `CHANEGLOG` and `package.json` was updated 
7. The CI try to rebase `beta` into `main`. The error occurred. `CHANGELOG` and `package.json` of docs got conflicts! 

Screenshot 2566-05-19 at 15 46 11


**Scenario 2** `docs` was not pre-release yet. But the team found that version `0.0.0` need a hotfix. 

1. Like scenario 1. Release hotfix 
2. The CI try to rebase `beta` into `main`. No conflicts. Everything works well. 

So, to summarize, you may have to manually rebase if you're in scenario 1. But, you don't have to do anything if you're in scenario 2. 

The pre-release is the hardest thing to do when you want to apply versioning system to your own repository, said by [Changeset themself](https://github.com/changesets/changesets/blob/main/docs/prereleases.md). So, I suggest that you don't have to add pre-release to your workflows if you don't have to. Just use `release-without-versioning` like `dev` for `beta`. 

For your PR, unfortunately, it do not solve the problem you raised. So, I'll close this with the comment.
